docs/examples/nmt/gnmt.md

```diff
@@ -0,0 +1,438 @@
+# Google NMT on IWSLT 2015 English-Vietnamese Translation
+
+In this notebook, we going to train Google NMT on IWSLT 2015 English-Vietnamese
+Dataset. The building prcoess includes: 1) load and process dataset and 2)
+create sampler and DataLoader and 3) build model and 4) write training epochs.
+
+## Load MXNET and Gluon
+
+```python
+import argparse
+import time
+import random
+import os
+import io
+import logging
+import numpy as np
+import mxnet as mx
+from mxnet import gluon
+from mxnet.gluon.data import ArrayDataset, SimpleDataset
+from mxnet.gluon.data import DataLoader
+import gluonnlp.data.batchify as btf
+from gluonnlp.data import ExpWidthBucket, FixedBucketSampler, IWSLT2015
+from gluonnlp.model import BeamSearchScorer
+from scripts.nmt.gnmt import get_gnmt_encoder_decoder
+from scripts.nmt.translation import NMTModel, BeamSearchTranslator
+from scripts.nmt.loss import SoftmaxCEMaskedLoss
+from scripts.nmt.utils import logging_config
+from scripts.nmt.bleu import compute_bleu
+import scripts.nmt._constants as _C
+```
+
+## Hyper-parameters
+
+```python
+np.random.seed(100)
+random.seed(100)
+mx.random.seed(10000)
+ctx = mx.gpu(0)
+
+# parameters for dataset
+dataset = 'IWSLT2015'
+src_lang = 'en'
+tgt_lang = 'vi'
+src_max_len = 50
+tgt_max_len = 50
+
+# parameters for model
```

Comment:
not related to this PR:
@szha maybe for gluonnlp we also want to define hyper_param registration like t2t for better abstraction? https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/models/transformer.py#L1467